tests/__tests__/fs/Fs_test.re

```diff
@@ -9,7 +9,12 @@ open TestFramework;
 
 external reraise: exn => _ = "%reraise";
 
-let unresolvedExePath = Fp.absoluteExn(Sys.executable_name);
+let normalizePathSeparator = {
+  let backSlashRegex = Str.regexp("\\\\");
+  pathStr => pathStr |> Str.global_replace(backSlashRegex, "/");
+};
+
+let unresolvedExePath = Fp.absoluteExn(Sys.executable_name |> normalizePathSeparator);
```

Comment:
This was also blocked Windows - the same issue that I set up for `Dir` - when we get a raw-windows-path-with-backslashes, we crash.

It makes me think that, maybe for now, we should update the `Fp.absoluteExn` API like:
```
let absoluteExn = (~normalizeBackSlashes=Sys.win32, path) => {

};
```

At least in the mean-time, it would work OK for both `Dir` and these test cases - and the normalization behavior could be overridden if desired. LMK what you think, @jordwalke

Reply:
I agree. Though I would be okay with making the complexity of cross platform paths front and center, at the time they parse the string. So that could be something like splitting the path parsing entrypoints into a couple of different named functions that make them acknowledge what they are doing. That is a little annoying, but the point is that once you get things into Paths, you never ever have to think about this stuff again.


```
let absoluteCurrentPlatformPathExn = (path) => {
....
};
let absolutePlatformPathExn = (~fromPlatform, path) => {
....
};
```
The second having no default argument. What do you think? Too annoying? These should be seldom used - but maybe in reality, it's not the case?

Reply:
Seems reasonable (although maybe a bit verbose) - but at least this is only needed at 'ingestion' of paths. And if you use `Dir` - you shouldn't really need to touch these very often (especially if we provide the `executingDirectory` and `currentWorkingDirectory` as well somewhere).

The only other issue I see is that right now `Fp` doesn't have a concept of `platform` - so we'll just have to bring that over from `Dir`. But seems OK! I'll give it a shot and see what you think.